### ISSUE TITLE
Need Example to Create Logical JBOD on D3940 (No RAID)

### ISSUE THREAD

**Issue by davew1234** [HewlettPackard/POSH-HPOneView#385]
### Expected Behavior
I can't find an example on how to create Logical JBODs (aka External logical JBOD) on the Synergy D3940 in a Server Profile Template. I can do this using OneView, but not using the POSH scripts.
I would like to create 8 External logical JBODs in a server profile template, to use for installing ESXi VSAN. I attached a screen shot of OneView that I created manually.

### Actual Behavior
This is as close as I can get. I tried many alternatives with parameters and the errors always say I have to specify RAID. I tried HBA mode but don't know how to specify disks.

```powershell
$DataDisk   = New-HPOVServerProfileLogicalDisk -Name DataDisk -DriveSelectionBy DriveType -DriveType SASSSD -MinDriveSize 1000 -MaxDriveSize 3000 -EraseDataOnDelete:$false -NumberOfDrives 8 -RAID RAID0 -StorageLocation External -Bootable $false

$DASstorage = New-HPOVServerProfileLogicalDiskController -ControllerID "Mezz 1" -Mode RAID -WriteCache Enabled -LogicalDisk $DataDisk
```

### Steps to reproduce
This is what I captured for the correct config that I created using OneView. I don't understand what parameters to use with POSH scripts to create this.

```powershell
PS C:\> $aa = Get-HPOVServerProfileTemplate
PS C:\> $aa | format-list

Name                 : SPT-Windows 2016 Server 480G10 v001
Status               : OK
Description          : Template Windows 2016 Server SY480 Gen10
Server Hardware Type : SY 480 Gen10 768GB M1 M3
Enclosure Group      : EG SFL B56 Lab01
Affinity             : Bay
Serial Number        :  (Physical)
UUID                 :  (Physical)
Connections          : 0
Manage BIOS          : True

PS C:\> $aa | select-object -property *

type                     : ServerProfileTemplateV5
uri                      : /rest/server-profile-templates/22730198-446a-4eca-a67b-9dd95c5eeeed
name                     : SPT-Windows 2016 Server 480G10 v001
description              : Template Windows 2016 Server SY480 Gen10
serverProfileDescription : Synergy 480 Gen10
serverHardwareTypeUri    : /rest/server-hardware-types/22605ACD-1B4F-45E2-877F-36852B33E853
enclosureGroupUri        : /rest/enclosure-groups/fffe427e-6742-40a9-8bfb-72087b08cb5e
affinity                 : Bay
hideUnusedFlexNics       : True
macType                  : Virtual
wwnType                  : Virtual
serialNumberType         : Physical
iscsiInitiatorNameType   : AutoGenerated
osDeploymentSettings     :
firmware                 : @{manageFirmware=True; firmwareInstallType=FirmwareOnlyOfflineMode;
                           forceInstallFirmware=False; firmwareBaselineUri=/rest/firmware-drivers/SPP_2018_09_20180919_
                           for_HPE_Synergy_Z7550-96560; firmwareActivationType=NotScheduled}
bootMode                 : @{manageMode=True; mode=UEFI; pxeBootPolicy=Auto; secureBoot=Unmanaged}
boot                     : @{manageBoot=True; order=System.Object[]}
bios                     : @{manageBios=True; overriddenSettings=System.Object[]}
localStorage             : @{sasLogicalJBODs=System.Object[]; controllers=System.Object[]}
sanStorage               : @{manageSanStorage=False; volumeAttachments=System.Object[]}
category                 : server-profile-templates
created                  : 2018-12-13T08:20:37.707Z
modified                 : 2018-12-13T08:22:53.983Z
status                   : OK
state                    :
scopesUri                : /rest/scopes/resources/rest/server-profile-templates/22730198-446a-4eca-a67b-9dd95c5eeeed
eTag                     : 1544689373983/2
connectionSettings       : @{manageConnections=True; connections=System.Object[]}
ApplianceConnection      : 10.27.36.200

PS C:\> $aa.localstorage | select-object -property * | format-list

sasLogicalJBODs : {@{id=1; deviceSlot=Mezz 1; name=DataDisk; numPhysicalDrives=8; driveMinSizeGB=1000;
                  driveMaxSizeGB=3000; driveTechnology=SasSsd; eraseData=False}}
controllers     : {@{deviceSlot=Mezz 1; mode=Mixed; initialize=False; driveWriteCache=Unmanaged; logicalDrives=}}
```

### Version Information
HPE OneView version: 4.10.03.364293.00
HPE REST POSH version: 4.10.1889.2173
PowerShell: 5.1.14409.1018

![error_jbod_disks](https://user-images.githubusercontent.com/37392057/49929802-3267e380-fe78-11e8-93a0-6f62cabec010.jpg)

**Comment by ChrisLynchHPE** [comment]
Sorry for getting to this thread so late.  Can you provide the exact error message you are getting?

**Comment by ChrisLynchHPE** [comment]
While I don't have a Synergy Gen10 device to test with, I was able to successfully create the local storage configuration you are trying to do with the Cmdlets above (I had to remove the controller cache setting as that isn't supported with the Gen9 I have):

![image](https://user-images.githubusercontent.com/6956305/51132898-5d7f8300-17e8-11e9-9a01-eaf871b63ad4.png)
![image](https://user-images.githubusercontent.com/6956305/51132923-6bcd9f00-17e8-11e9-97b6-7a0b98512a75.png)

**Comment by ChrisLynchHPE** [comment]
After further analysis, it appears that the library has a few bugs that are preventing this configuration from being successful.  The following two examples are supposed to support what you are looking for

```powershell
# Using Size and Technology for drive selection
$logicaldisk1 = New-HPOVServerProfileLogicalDisk -Name jobd1 -NumberofDrives 4 -DriveType SASSSD -DriveSelectionBy SizeAndTechnology -MinDriveSize 4000 -MaxDriveSize 4000

# Get available drives and use to request for logical disk
$AvailableDriveType = Get-HPOVSasLogicalInterconnect -Name "LE1-Default SAS Synergy LIG-3" -ErrorAction Stop | Get-HPOVAvailableDriveType  | ? { $_.Type -eq "SASHDD" -and $_.capacity -eq 900 }
$LogicalDisk1 = New-HPOVServerProfileLogicalDisk -Name jbod1 -DriveSelectionBy DriveType -NumberofDrives 6 -AvailableDriveType $AvailableDriveType
```

I have performed initial tests that are successful in a fix.  I will need to perform full regression testing to make sure there are no other impactful changes to this fix.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 4.10.1954.1523](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v4.10.1954.1523) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPOneView.410/4.10.1954.1523).

Please verify before this issue is closed.

**Comment by ChrisLynchHPE** [comment]
Closing do to no further activity.
